spec/requests/donations_requests_spec.rb

```diff
@@ -172,5 +172,77 @@
+        TestInventory.create_inventory(organization, { storage_location.id => { item.id => 0, extra_item.id => 1 } })
+        original_quantity = 100
+        original_source = Donation::SOURCES[:manufacturer]
+        original_date = DateTime.new(2024)
+        donation = create(:manufacturer_donation, :with_items, item: item, item_quantity: original_quantity, issued_at: original_date, organization: organization, storage_location: storage_location, source: original_source)
+        distribution = {
+          storage_location_id: storage_location.id,
+          partner_id: create(:partner).id,
+          delivery_method: :delivery,
+          line_items_attributes: {
+            "0": { item_id: item.id, quantity: 90 }
+          }
+        }
+
+        post distributions_path(distribution: distribution, format: :turbo_stream)
+
+        edited_source = Donation::SOURCES[:product_drive]
+        edited_source_drive = create(:product_drive, organization: organization)
+        edited_source_drive_participant = create(:product_drive_participant, organization: organization)
+        edited_storage_location = create(:storage_location, name: "Test Storage", organization: organization)
+        edited_money = 10.0
+        edited_comment = "New test comment"
+        edited_date = "2019-01-01"
+        extra_quantity = 1
+        edited_quantity = 1
+
+        edited_donation = {
+          source: edited_source,
+          product_drive_id: edited_source_drive.id,
+          product_drive_participant_id: edited_source_drive_participant.id,
+          storage_location_id: edited_storage_location.id,
+          money_raised_in_dollars: edited_money,
+          comment: edited_comment,
+          issued_at: edited_date,
+          line_items_attributes: {
+            "0": { item_id: item.id, quantity: edited_quantity },
+            "1": { item_id: extra_item.id, quantity: extra_quantity }
+          }
+        }
+
+        put donation_path(id: donation.id, donation: edited_donation)
+
+        if Event.read_events?(organization)
+          expect(flash[:alert]).to include("Error updating donation: Could not reduce quantity")
+        else # TODO remove this branch when switching to events
+          expect(flash[:alert]).to include("Error updating donation: Requested items exceed the available inventory")
+        end
+
+        expect(response.body).to include("Edit - Donations - #{original_source}")
+        expect(response.body).to include("Editing Donation\n          <small>from #{original_source}")
+        expect(response.body).to include("<li class=\"breadcrumb-item\">\n            <a href=\"#\">Editing #{original_source}")
+        expect(response.body).to include("<option selected=\"selected\" value=\"#{edited_source}\">#{edited_source}</option>")
+        expect(response.body).to include("<option selected=\"selected\" value=\"#{edited_source_drive.id}\">#{edited_source_drive.name}</option>")
```

Comment:
I'd prefer to specify the name etc. (anything we're actually checking) when creating the data. This makes the tests less flaky and more easily reproducible.

Reply:
resolved by [44a6ef4](https://github.com/rubyforgood/human-essentials/pull/4483/commits/44a6ef42107f6355b6d6f8f38286b4b4a6ea1103)